larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewThreeDKinkTool.cc

```diff
@@ -0,0 +1,479 @@
+                modification.m_affectedClusters.push_back(particle.m_pClusterB);
+                modification.m_affectedClusters.push_back(particle.m_pCommonCluster);
+
+                modificationList.push_back(modification);
+            }
+            catch (const StatusCodeException &statusCodeException)
+            {
+                if (STATUS_CODE_FAILURE == statusCodeException.GetStatusCode())
+                    throw statusCodeException;
+
+                continue;
+            }
+        }
+    }
+}
+
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TwoViewThreeDKinkTool::IsThreeDKink(TwoViewTransverseTracksAlgorithm *const pAlgorithm, const Particle &particle,
+    const CartesianVector &splitPosition, const bool isALowestInX) const
+{
+    try
+    {
+        const TwoDSlidingFitResult &fitResultCommon(pAlgorithm->GetCachedSlidingFitResult(particle.m_pCommonCluster));
+        const TwoDSlidingFitResult &lowXFitResult(isALowestInX ? pAlgorithm->GetCachedSlidingFitResult(particle.m_pClusterA)
+                                                               : pAlgorithm->GetCachedSlidingFitResult(particle.m_pClusterB));
+        const TwoDSlidingFitResult &highXFitResult(isALowestInX ? pAlgorithm->GetCachedSlidingFitResult(particle.m_pClusterB)
+                                                                : pAlgorithm->GetCachedSlidingFitResult(particle.m_pClusterA));
+
+        const float minusX(this->GetXSamplingPoint(splitPosition, false, lowXFitResult, fitResultCommon));
+        const float plusX(this->GetXSamplingPoint(splitPosition, true, highXFitResult, fitResultCommon));
+        const float splitX(splitPosition.GetX());
+
+        CartesianVector minus1(0.f, 0.f, 0.f), split1(0.f, 0.f, 0.f), plus1(0.f, 0.f, 0.f);
+        CartesianVector minus2(0.f, 0.f, 0.f), split2(splitPosition), plus2(0.f, 0.f, 0.f);
+
+        if ((STATUS_CODE_SUCCESS != fitResultCommon.GetGlobalFitPositionAtX(minusX, minus1)) ||
+            (STATUS_CODE_SUCCESS != fitResultCommon.GetGlobalFitPositionAtX(splitX, split1)) ||
+            (STATUS_CODE_SUCCESS != fitResultCommon.GetGlobalFitPositionAtX(plusX, plus1)) ||
+            (STATUS_CODE_SUCCESS != lowXFitResult.GetGlobalFitPositionAtX(minusX, minus2)) ||
+            (STATUS_CODE_SUCCESS != highXFitResult.GetGlobalFitPositionAtX(plusX, plus2)))
+        {
+            return m_splitMode; // split mode rules, by default
+        }
+
+        // Extract results
+        const HitType hitType1(LArClusterHelper::GetClusterHitType(particle.m_pCommonCluster));
+        const HitType hitType2(LArClusterHelper::GetClusterHitType(particle.m_pClusterA));
+
+        CartesianVector minus(0.f, 0.f, 0.f), split(0.f, 0.f, 0.f), plus(0.f, 0.f, 0.f);
+        float chi2Minus(std::numeric_limits<float>::max()), chi2Split(std::numeric_limits<float>::max()),
```

Comment:
Given you don't use the chi2 values, could just define one dummyChi2 variable and reuse that in all three cases.